src/meili/map_matcher.cc

```diff
@@ -664,12 +697,15 @@ std::vector<MatchResults> MapMatcher::OfflineMatch(const std::vector<Measurement
       const auto& this_stateid = original_state_ids[time];
       const auto& next_stateid =
           time + 1 < original_state_ids.size() ? original_state_ids[time + 1] : StateId();
+
+      midgard::PointLL first_projection = results[time].lnglat;
+      midgard::PointLL last_projection = results[time + 1].lnglat;
       const auto& interpolated_results =
-          InterpolateMeasurements(*this, it->second, this_stateid, next_stateid);
+          InterpolateMeasurements(*this, it->second, this_stateid, next_stateid, first_projection,
+                                  last_projection);
 
       // Copy the interpolated match results into the final set
-      std::copy(interpolated_results.cbegin(), interpolated_results.cend(),
-                std::back_inserter(best_path));
+      best_path.insert(best_path.cend(), interpolated_results.cbegin(), interpolated_results.cend());
```

Comment:
since the results are a non trivial class perhaps you should use a move iterator

Reply:
if it does not manage resource, the move operator is by default equals to the copy constructor. I checked the struct MatchResult. None of the members are managing resources actually. Then the move is a copy anyway.

Reply:
good point, i couldnt remember all the members of the class but i guess its just a collection of integral types